### PR TITLE
Make mobile worker workflow location safe

### PR DESCRIPTION
**Pull request by esoergel**
@millerdev

### CODE REVIEW

File: corehq/apps/users/static/users/js/mobile_workers.ng.js [hunk 1]
```diff
@@ -341,6 +341,18 @@
         };
     };
 
+    mobileWorkerDirectives.validateLocation = function ($http, $q, djangoRMI) {
```
Comment (dimagimon) [inline corehq/apps/users/static/users/js/mobile_workers.ng.js]: 'djangoRMI' is defined but never used (no-unused-vars)

File: corehq/apps/users/views/mobile/users.py [hunk 2]
```diff
@@ -136,11 +147,12 @@ def custom_data(self):
     def editable_user(self):
         try:
             user = CommCareUser.get_by_user_id(self.editable_user_id, self.domain)
-            if not user:
-                raise Http404()
-            return user
         except (ResourceNotFound, CouchUser.AccountTypeError, KeyError):
             raise Http404()
+        else:
```
Comment (millerdev) [inline corehq/apps/users/views/mobile/users.py]: `else:` not necessary (block could be de-indented)

**Comment by millerdev** [comment]
I didn't notice any tests. Would it make sense to add some?

File: corehq/apps/users/views/mobile/users.py (hunk 2 above)
Comment (esoergel) [inline corehq/apps/users/views/mobile/users.py]: good point.  Also, on further inspection, I'm having a hard time figuring out how user could ever be falsy.

**Comment by esoergel** [comment]
doh, yes, yes it would

**Pull request merged**